### PR TITLE
22 new data source security account

### LINKED ISSUE #22: [New Data Source]: security_account
### Description

Should support

name
Application
Owner.name
role
password
security_authentication_method
comment
locked
ldap_fastbind

### Requested Resource(s) and/or Data Source(s)

* netapp_ontap_security_account_data_source

### Potential Terraform Configuration

_No response_

### References

https://library.netapp.com/ecmdocs/ECMLP2885799/html/#/security/account_collection_get

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by carchi8py**
Adds #22

### CODE REVIEW

File: internal/interfaces/security_account.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/mitchellh/mapstructure"
+	"github.com/netapp/terraform-provider-netapp-ontap/internal/restclient"
+	"github.com/netapp/terraform-provider-netapp-ontap/internal/utils"
+)
+
+// TODO:
```
Comment (chuyich) [inline internal/interfaces/security_account.go]: These lines can be removed.

File: internal/provider/security_account_data_source.go [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "SecurityAccount name",
+				Required:            true,
+			},
+			"id": schema.StringAttribute{
+				MarkdownDescription: "SecurityAccount id",
+				Computed:            true,
+			},
+			"owner": schema.SingleNestedAttribute{
```
Comment (chuyich) [inline internal/provider/security_account_data_source.go:86]: 'owner' should be required in one data source case, right?

File: internal/provider/security_accounts_data_source.go [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/netapp/terraform-provider-netapp-ontap/internal/interfaces"
+	"github.com/netapp/terraform-provider-netapp-ontap/internal/utils"
+)
+
+// TODO:
```
Comment (chuyich) [inline internal/provider/security_accounts_data_source.go]: these lines can be removed

File: docs/data-sources/security_account_data_source.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+- `locked` (Boolean) SecurityAccount locked
+- `role` (Attributes) SecurityAccount role (see [below for nested schema](#nestedatt--role))
+- `scope` (String) SecurityAccount scope
+
+<a id="nestedatt--owner"></a>
+### Nested Schema for `owner`
+
+Required:
+
+- `name` (String) SecurityAccount owner name
```
Comment (chuyich) [inline docs/data-sources/security_account_data_source.md:53]: If the user only provides "name", then it will be multiple records.

**Review by chuyich — CHANGES_REQUESTED** [review]
One question on the single data source: the user needs to provide name and owner info to get the single record. So owner probably needs to be required.

**Review by chuyich — CHANGES_REQUESTED** [review]
also missing the CHANGELOG update

File: internal/provider/security_account_data_source.go (hunk 2 above)
Comment (carchi8py) [reply internal/provider/security_account_data_source.go:86]: NO.... YES.....

So an account can be SVM specific or cluster specific. 

Let me try some test out, my worry here is cluster name might not be known by the user.

File: docs/data-sources/security_account_data_source.md (hunk 4 above)
Comment (carchi8py) [reply docs/data-sources/security_account_data_source.md:53]: I don't think there a way around that.

**Comment by carchi8py** [comment]
> One question on the single data source: the user needs to provide name and owner info to get the single record. So owner probably needs to be required.

Since cluster user can't give an owner info we can't have it required.

File: scripts/acctest.sh [hunk 5]
```diff
@@ -2,11 +2,12 @@
 
 org_dir=$(pwd)
 export TF_ACC=1
-#export TF_ACC_NETAPP_HOST="<Host1>"
-#export TF_ACC_NETAPP_HOST2="<host2>>"
-#export TF_ACC_NETAPP_USER="admin"
-#export TF_ACC_NETAPP_PASS="<password>"
-#export TF_ACC_NETAPP_LICENSE="<licensekey>"
+export TF_ACC_NETAPP_HOST="10.193.180.108"
```
Comment (chuyich) [inline scripts/acctest.sh]: Is it ok that we hard-coded these values?

**Comment by chuyich** [comment]
Please resolve the conflict, then it should be good to go.

**Review by chuyich — APPROVED** [review]

**Pull request merged**